src/commands/run.rs

```diff
@@ -0,0 +1,263 @@
+//! Implementation of the run command.
+
+use std::fs;
+use std::path::Path;
+use std::path::PathBuf;
+use std::path::absolute;
+
+use anyhow::Context;
+use anyhow::Result;
+use anyhow::anyhow;
+use anyhow::bail;
+use chrono;
+use clap::Parser;
+use codespan_reporting::files::SimpleFile;
+use codespan_reporting::term::emit;
+use tracing_log::log;
+use url::Url;
+use wdl::analysis::path_to_uri;
+use wdl::engine::Engine;
+use wdl::engine::EvaluationError;
+use wdl::engine::Inputs;
+use wdl::engine::local::LocalTaskExecutionBackend;
+use wdl::engine::v1::TaskEvaluator;
+
+use crate::Mode;
+use crate::analyze;
+use crate::get_display_config;
+
+/// Arguments for the run command.
+#[derive(Parser, Debug)]
+#[command(author, version, about)]
+pub struct RunArgs {
+    /// The path or URL to the WDL document containing the task to run.
+    #[arg(value_name = "PATH or URL")]
+    pub file: String,
+
+    /// The path to the input JSON file.
+    ///
+    /// If not provided, an empty set of inputs will be sent to the task.
+    #[arg(short, long, value_name = "JSON", conflicts_with = "name")]
+    pub inputs: Option<PathBuf>,
+
+    /// The name of the task to run.
+    ///
+    /// Required if no `inputs` file is provided.
+    #[arg(short, long, value_name = "NAME", conflicts_with = "inputs")]
+    pub name: Option<String>,
+
+    /// The output directory; defaults to the task name.
+    ///
+    /// If no output directory is provided, a default nested directory is
+    /// created based on the task name and the current time in the form
+    /// `sprocket_runs/<task_name>/<timestamp>/`.
```

Comment:
```suggestion
    /// `sprocket_runs/<execution_name>/<timestamp>/`.
```
I think this one should definitely be written as the more generic executable, since we're establishing a directory structure and it will need to support task and workflow execution.

Reply:
Agree, that's a good catch.